### PR TITLE
Point master branch of rpc-openstack at head of liberty branch of OSA

### PR DESCRIPTION
**Pull request by mancdaz**
Point master branch of rpc-openstack at head of liberty branch of OSA

In order to begin development towards a liberty release, we need to
switch our master branch to point at osa liberty. This commit bumps
forward to the head of osa liberty

12.0.5-4-ga91fa43 (12.0.5 plus 4 commits)

This commit also contains necessary changes to have the gate run and
pass linting, including:

- use ephemeral disk on cloud server when AIO
- bootstrap ansible before bootstrap aio (which in liberty is ansible)
- run lint using test-requirements.txt instead of dev-requirements.txt
- pass a list of locations that should be scanned for files containing
  information about what packages to build (allows us to define
  /opt/rpc-openstack/rpcd so that all of the rpc packages can be built at
  the same time by the osa repo-builder and we don't need separate rpc
  plays to build our own packages)
- correct the location of the horizon-manage.py file due to venv use
- pin setuptools==19.3 to allow holland to install
- pin elasticsearch<2.1.0 as this is a requirement of latest
  elasticsearch-curator

NOTE: I have not included the patch to make the holland wheels get built, rather than installing locally.  Instead, I have gone with the less intrusive option to pin setuptools and allow the old method to work.  Since this PR is an effort to do the minimum to get a working gate for liberty, I felt that work could be done in a subsequent patch

### CODE REVIEW

**Comment by hughsaunders** [comment]
recheck

**Comment by hughsaunders** [comment]
This is a genuine failure - liberty includes the ansibleised aio bootstrap, and deploy.sh will need to be updated to call bootstrap-ansible before bootstrap-aio

**Comment by mancdaz** [comment]
@hughsaunders noted -  that's why I made it a WIP while I worked on that

**Comment by mancdaz** [comment]
currently failing to build the rpc_repo bits:

```
13:05:10 TASK: [Create a build report for all known packages within a release] ********* 
13:05:10 failed: [jrpcaio-98_repo_container-ee320f92] => (item={'packages': ['beaver==33.3.0', 'elasticsearch-curator', 'ipaddr', 'rackspace-monitoring-cli'], 'remote_package_parts': [], 'remote_packages': [], 'role_packages': {'beaver': ['beaver==33.3.0'], 'elasticsearch': ['elasticsearch-curator'], 'rpc_maas': ['ipaddr', 'rackspace-monitoring-cli']}}) => {"changed": true, "cmd": "yaprt --quiet create-report --report-file /var/www/repo/reports/release-rpc-r11.1.0rc5-report.json --git-install-repos [] --packages ['beaver==33.3.0', 'elasticsearch-curator', 'ipaddr', 'rackspace-monitoring-cli']", "delta": "0:00:02.108453", "end": "2016-01-19 13:05:08.996067", "item": {"packages": ["beaver==33.3.0", "elasticsearch-curator", "ipaddr", "rackspace-monitoring-cli"], "remote_package_parts": [], "remote_packages": [], "role_packages": {"beaver": ["beaver==33.3.0"], "elasticsearch": ["elasticsearch-curator"], "rpc_maas": ["ipaddr", "rackspace-monitoring-cli"]}}, "rc": 1, "start": "2016-01-19 13:05:06.887614", "warnings": []}
13:05:10 stderr: fatal: repository '[]' does not exist
13:05:10 
13:05:10 FATAL: all hosts have already failed -- aborting
13:05:10 
13:05:10 PLAY RECAP ******************************************************************** 
13:05:10            to retry, use: --limit @/jenkins/repo-build.retry
13:05:10 
13:05:10 jrpcaio-98_repo_container-ee320f92 : ok=1    changed=0    unreachable=0    failed=1   
```

because of this flag being passed to yaprt:

```
--git-install-repos []
```

**Comment by mancdaz** [comment]
spoke to @odyssey4me - we need to change the way rpc specific wheels are built due to the changes in wheel building in liberty. The new process should allow us to have all wheels (both osa and rpc) built in the same repo

**Comment by mancdaz** [comment]
So current status:

With this patch on top of osa https://review.openstack.org/#/c/271233/, the build completes almost successfully. It's currently failing on rpc bits, specifically installing holland and plugins:

```
failed: [liberty_galera_container-5fb53721] => {"ansible_job_id": "399401095020.8734", "attempts": 5, "changed": false, "cmd": "/usr/local/bin/pip install /opt/holland_v1.0.10", "failed": true, "finished": 1}
msg: Task failed as maximum retries was encountered
failed: [liberty_galera_container-157d0876] => {"ansible_job_id": "399401095020.9389", "attempts": 5, "changed": false, "cmd": "/usr/local/bin/pip install /opt/holland_v1.0.10", "failed": true, "finished": 1}
msg: Task failed as maximum retries was encountered
failed: [liberty_galera_container-8a1ebbf7] => {"ansible_job_id": "399401095020.8327", "attempts": 5, "changed": false, "cmd": "/usr/local/bin/pip install /opt/holland_v1.0.10", "failed": true, "finished": 1}
msg: Task failed as maximum retries was encountered
<job 399401095020.9389> FAILED on liberty_galera_container-157d0876
<job 399401095020.8734> FAILED on liberty_galera_container-5fb53721
<job 399401095020.8327> FAILED on liberty_galera_container-8a1ebbf7
```

Logging in and attempting to install holland manually:

```
root@liberty_galera_container-157d0876:/opt# pip install holland_v1.0.10/
Ignoring indexes: https://pypi.python.org/simple
Processing ./holland_v1.0.10
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 3, in <module>
      File "/usr/local/lib/python2.7/dist-packages/setuptools/__init__.py", line 11, in <module>
        from setuptools.extern.six.moves import filterfalse, map
      File "/usr/local/lib/python2.7/dist-packages/setuptools/extern/__init__.py", line 1, in <module>
        from pkg_resources.extern import VendorImporter
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3130, in <module>
        @_call_aside
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3116, in _call_aside
        f(*args, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3155, in _initialize_master_working_set
        add_activation_listener(lambda dist: dist.activate())
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 956, in subscribe
        callback(dist)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3155, in <lambda>
        add_activation_listener(lambda dist: dist.activate())
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2655, in activate
        declare_namespace(pkg)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2225, in declare_namespace
        _handle_ns(packageName, path_item)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2194, in _handle_ns
        path.sort(key=sort_key)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2192, in sort_key
        return sys_path.index(_normalize_cached(os.sep.join(parts)))
    ValueError: '/tmp/pip-dCewZJ-build' is not in list

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-dCewZJ-build
```

currently troubleshooting that

**Comment by mancdaz** [comment]
so the issue currently being seen (above) isn't liberty specific - it's to do with a recent release of setuptools, and how we install holland. See https://github.com/rcbops/rpc-openstack/issues/777 for tracking of this issue.

We can fix this by actually building wheels for the holland packages, instead of trying to pip install against local git clone, which is what's currently breaking.

**Comment by sigmavirus24** [comment]
@mancdaz do you mean executing `python setup.py bdist_wheel`?

**Comment by mancdaz** [comment]
@sigmavirus24 no, I mean we hook in to the new package building process in liberty and have it build wheels for the holland bits and drop them into the repo. Then we can just pip install from the wheel repo like we do for all the other wheels.

This patch in OSA gave us the ability to make the osa repo builder build packages for rpc stuff:

https://github.com/openstack/openstack-ansible/commit/15b826b88aace97562811b5e6091b08044980ad2

instead of us having to run separate plays in rpco to build rpc packages, and you'll see as part of this commit I'm making use of that. But we are not even installing holland from a wheel in the repo, we're not winning with that.

At the moment, for holland, we don't do that. We clone the holland repo on each galera container, then execute a ```pip install /opt/holland-1.0.10``` which is the functionality that's broken in setuptools 19.4

**Comment by sigmavirus24** [comment]
So I think this is actually something the Holland developers need notification of too. We're likely not the only people that are going to run into this. They're relying on namespace packages for a reason which is not presently clear to me. We should file a bug against their project as well to determine why they think they need namespace packages (which have caused OpenStack endless headaches) and how we can help them remove that dependency.

**Comment by sigmavirus24** [comment]
@mancdaz what version of pip are you installing?

**Comment by d34dh0r53** [comment]
recheck

**Comment by mancdaz** [comment]
@sigmavirus24 pip 7.1.2, as fixed by osa. Tried also with 8.0.2 and 6.0.0 just for fun. Same issue.

Should probably discuss this over in https://github.com/rcbops/rpc-openstack/issues/777 now

**Comment by d34dh0r53** [comment]
:+1: remove the [WIP] from the title.

**Comment by mancdaz** [comment]
@d34dh0r53 thanks -  once I get a fix/workaround going for the setuptools/holland issue, this should pass gate. Until then it's not going to pass, so it's not ready for merge (hence WIP)

**Comment by mancdaz** [comment]
recheck

**Comment by hughsaunders** [comment]
recheck

**Comment by mancdaz** [comment]
@d34dh0r53 @prometheanfire and anyone else who wants to now test this - I've removed the WIP label as I'm now confident this is working (as tested locally - still waiting on jenkins).

NOTE: I have not included the patch to make the holland wheels get built, rather than installing locally.  Instead, I have gone with the less intrusive option to pin setuptools and allow the old method to work.  Since this PR is an effort to do the minimum to get a working gate for liberty, I felt that work could be done in a subsequent patch.

**Comment by mancdaz** [comment]
gate failed on setup-maas. Will investigate

**Comment by prometheanfire** [comment]
yep, tepest passed, maas couldn't find it's config

File: scripts/deploy.sh [hunk 1]
```diff
@@ -142,10 +152,10 @@ fi
 cd ${RPCD_DIR}/playbooks/
 
 # build the RPC python package repository
-run_ansible repo-build.yml
+#run_ansible repo-build.yml
```
Comment (prometheanfire) [inline scripts/deploy.sh]: should we just remove these instead of commenting out, this is git after all :P

Comment (mancdaz) [inline scripts/deploy.sh]: @pronetheanfire yeah, we can also remove the plays as we're not using them any more. Again, was trying to keep the patch size down but I guess in this case it should be done.

Comment (prometheanfire) [inline scripts/deploy.sh]: I'd be fine with a follow on patch, as long as it's quick

**Comment by hughsaunders** [comment]
Missing keystone_service_internalurl_v3 is related to this patch as the master periodics are passing.

**Comment by d34dh0r53** [comment]
recheck

**Comment by hughsaunders** [comment]
recheck

**Comment by mancdaz** [comment]
closing in favour of https://github.com/rcbops/rpc-openstack/pull/796

**Pull request closed without merging**